app/src/main/java/com/nervousfish/nervousfish/cryptography/KeyGenerator.java

```diff
@@ -0,0 +1,62 @@
+package com.nervousfish.nervousfish.cryptography;
+
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPrivateKeySpec;
+import java.security.spec.RSAPublicKeySpec;
+
+public class KeyGenerator {
+
+    /**
+     * Generates a random KeyPair with the RSA algorithm.
+     * @return a randomly generated KeyPair
+     */
+    public static KeyPair generateRandomKeyPair() {
+        try {
+            KeyPairGenerator k = KeyPairGenerator.getInstance("RSA");
```

Comment:
It should be final, otherwise it will error on the new Checkstyle rules (and also because it's good coding style because it's easier to reason about the program)